### ISSUE TITLE
ISSUE: Z-Wave integration not found

### ISSUE THREAD

**Issue by Trolann** [FutureTense/keymaster#226]
**Describe the bug**
Attempting to change codes does not work. Stalls at 'Adding...'

**Environment (please complete the following information):**

- OS: HassOS
- Type of system that HA is running on: Intel Windows 10
- Home Assistant version: 2021.8.4
- Hassio/Docker/Core? VirtualBox
- Component version: 0.0.63
- Z-Wave integration name: zwave js
- Lock make and model: Schlage Connect

**Logs**
2021-09-08 14:47:22 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found

**Screenshots**
![image](https://user-images.githubusercontent.com/72931206/132590508-d23c4b40-10cd-47d4-8cf3-b2f63e21777c.png)

**Additional context**

**Comment by Trolann** [comment]
All of my code slot sensors are also unavailable:

![image](https://user-images.githubusercontent.com/72931206/132591721-d3970c74-f511-423f-93a0-eaa5cd4be149.png)

**Comment by raman325** [comment]
I feel like the error message is pretty clear - how will the integration control your zwave locks if it can't find the zwave integration?

Do you have multiple zwave integrations installed by any chance? like zwave + zwave_js

**Comment by Trolann** [comment]
> I feel like the error message is pretty clear - how will the integration control your zwave locks if it can't find the zwave integration?
> 
> Do you have multiple zwave integrations installed by any chance? like zwave + zwave_js

I feel like I never said the error message wasn't clear - I know if it can't find the zwave integration it can't control my locks, that's why I raised the issue.

No, I only have zwave js and this install has never had zwave.

Updated to 2021.9.4 fixed the issue. Prior to the upgrade all code clots were unavailable and there was no binary_sensor for the <doorname>_network. Now all sensors are available.

Leaving open in case it warrants any further investigation, but it's solved on my side.

**Comment by raman325** [comment]
you're right, that tone was a bit over the top, I apologize. Your description of the bug seemed to be pointing to the wrong problem and I wrote that in a bit of a hurry.

As to updating to 2021.9.4 the only thing I can think of is that the version of keymaster you were using was for 2021.9 while you were on 2021.8 - there were breaking changes in the zwave-js library that had to be accounted for. But the HACS installation should require 2021.9 as the minimum version. Did you use HACS to install or did you install it manually? Just trying to understand what could have gotten you into that state

**Comment by djaesthetic** [comment]
@raman325 : Not to muddy the waters, but I'm experiencing a (_near?_) identical issue as @Trolann . The only difference with my config is a newer HA version **(2021.9.5)**. I'm running the same component ver, Zwave implementation, and lock type.

- Component **0.0.63**
- ZWaveJS (_connected via ZwaveJS2MQTT, in my case_)
- Schlage Connect lock (**BE469ZP**)

Previously I had FutureTense's "lock-manager" successfully running w/o issue. Over the weekend I wiped everything and moved to a completely fresh install. Lock/Unlock works fine, however I've gone through the _(now "Keymaster")_ install a half dozen times, each time resulting in my **PIN Status** stalling at _"**Adding"**_. All of my "_frontdoor: Code Slot 1_" code slot sensors are _also_ unavailable, same as @Trolann . I used the HACS install.

Coincidence? Absolutely possible, but unfortunately I'm already one version ahead from where they just moved to so there's no upgrade path for me to throw at the issue.

**Comment by raman325** [comment]
Do you see anything in the logs related to `keymaster` that could point to the issue?

**Comment by djaesthetic** [comment]
The only thing I see in the logs referencing "keymaster" at all is the boilerplate:

**2021-09-13 00:32:02 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration keymaster which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant**

I attempted to throttle logging way up to DEBUG but my novice eyes didn't see anything noteworthy. Just regular calls.

**Comment by firstof9** [comment]
Navigate to Dev-tools -> Services tab in Home Assistant. Select the service `logger.set_level` in data enter `custom_components.keymaster: debug` and press Call Service.

Your log will now start showing debug data in the Home Assistant log.

**Comment by rbauman70** [comment]
I'm having the same problem--unable to set/clear lock entries using keymaster.  Here are the log entries:

2021-09-18 10:55:10 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration keymaster which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-09-18 10:55:14 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found
2021-09-18 10:55:14 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found
2021-09-18 10:55:14 WARNING (MainThread) [homeassistant.components.binary_sensor] Platform keymaster not ready yet: None; Retrying in background in 30 seconds
2021-09-18 10:55:14 WARNING (MainThread) [homeassistant.components.binary_sensor] Platform keymaster not ready yet: None; Retrying in background in 30 seconds
2021-09-18 10:55:14 ERROR (MainThread) [custom_components.keymaster] Error fetching keymaster data:
2021-09-18 10:55:14 ERROR (MainThread) [custom_components.keymaster] Error fetching keymaster data:
2021-09-18 10:55:27 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found
2021-09-18 10:55:27 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found
2021-09-18 10:56:27 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found
2021-09-18 10:56:27 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found

Keymaster was working correctly when I installed it 6 weeks ago, but some update has broken it.  Current versions:
HA core-2021.8.8
zwavejs2mqtt: 5.6.0
zwave-js: 8.3.1
keymaster: v0.0.63

I can call the zwave_js.set_lock_usercode service and it is setting the lock codes correctly.  The lock/unlock function is working correctly.  All other Z-Wave devices appear to be functioning correctly.   I've deleted and re-installed the keymaster integration using HACS for the two locks I have.  No change.
